### PR TITLE
PS binlog decryption implemented. I have not touched the mysqlbinlog

### PR DESCRIPTION
**Pull request by robgolebiowski**
part as this will stay as is in PS.

### CODE REVIEW

File: libbinlogevents/include/binlog_event.h [hunk 1]
```diff
@@ -789,7 +793,7 @@ class Binary_log_event {
      is not to be handled, it does not exist in binlogs, it does not have a
      format).
   */
-  static const int LOG_EVENT_TYPES = (ENUM_END_EVENT - 1);
+  static constexpr int LOG_EVENT_TYPES = (MYSQL_END_EVENT - 1);
```
Comment (laurynas-biveinis) [inline libbinlogevents/include/binlog_event.h]:
```suggestion
  static const constexpr int LOG_EVENT_TYPES = (MYSQL_END_EVENT - 1);
```

File: mysql-test/suite/rpl/t/percona_rpl_encryption_slave_binlog_ps_encrypted.test [hunk 2]
```diff
@@ -0,0 +1,69 @@
+--source include/have_debug.inc
+--source include/have_binlog_format_row.inc
+--let $rpl_skip_start_slave= 1
+--source include/master-slave.inc
+
+# The test us done with prepared PS encrypted relay logs (encryption key is in
```
Comment (laurynas-biveinis) [inline mysql-test/suite/rpl/t/percona_rpl_encryption_slave_binlog_ps_encrypted.test]:
```suggestion
# The test is done with prepared PS encrypted relay logs (encryption key is in
```

File: mysql-test/suite/rpl/t/percona_rpl_encryption_slave_binlog_ps_encrypted.test [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+--let $rpl_server_number= 2
+--let $rpl_server_parameters= $keyring_parameters
+--let $rpl_omit_print_server_parameters= 1
+--source include/rpl_start_server.inc
+--source include/start_slave.inc
+
+--source include/rpl_connection_master.inc
+--source include/sync_slave_sql_with_master.inc
+
+--let $assert_text= Slave shall have t1 with a 5 rows on it
```
Comment (laurynas-biveinis) [inline mysql-test/suite/rpl/t/percona_rpl_encryption_slave_binlog_ps_encrypted.test:61]: rpl_diff.inc ?

**Review by laurynas-biveinis — CHANGES_REQUESTED** [review]
Might be a good idea to rename our  START_ENCRYPTION_EVENT source code mnemonics to something implying being not-current (ENCRYPTION57 ?).

In the testcase, can you do 1->2->3, and check on 3 that no Percona binlog encryption is present in the relay log?

**Comment by robgolebiowski** [comment]
I will implement "In the testcase, can you do 1->2->3, and check on 3 that no Percona binlog encryption is present in the relay log?" when I will be merging this to PS. The reason being that I need mysqlbinlog which recognizes START_ENCRYPTION_EVENT in order to be able to make sure it does not occur (in relay log). Otherwise I could do this with DBUG variable. But mysqlbinlog would make code cleaner ?

**Comment by robgolebiowski** [comment]
ready for 2nd round

File: sql/event_crypt.h [hunk 4]
```diff
@@ -0,0 +1,12 @@
+#ifndef EVENT_ENCRYPTER_H
+#define EVENT_ENCRYPTER_H
+
+#include "basic_ostream.h"
```
Comment (laurynas-biveinis) [inline sql/event_crypt.h:4]: Please check this header list for pruning too: basic_ostream, rpl_constants, binlog_event?

Comment (robgolebiowski) [reply sql/event_crypt.h:4]: Will do, after the merge is completed I assume ?

Comment (laurynas-biveinis) [reply sql/event_crypt.h:4]: Yep

**Pull request closed without merging**